controllers/replication.storage/pvc.go

```diff
@@ -56,3 +58,36 @@ func (r VolumeReplicationReconciler) getPVCDataSource(logger logr.Logger, req ty
 
 	return pvc, pv, nil
 }
+
+// annotatePVCWithOwner will add the VolumeReplication details to the PVC annotations.
+func (r *VolumeReplicationReconciler) annotatePVCWithOwner(ctx context.Context, logger logr.Logger, req types.NamespacedName, pvc *corev1.PersistentVolumeClaim) error {
+	if pvc.ObjectMeta.Annotations == nil {
+		pvc.ObjectMeta.Annotations = map[string]string{}
+	}
+
+	ownerName := pvc.ObjectMeta.Annotations[replicationv1alpha1.VolumeReplicationNameAnnotation]
+	if ownerName == "" {
+		pvc.ObjectMeta.Annotations[replicationv1alpha1.VolumeReplicationNameAnnotation] = req.Name
+		err := r.Update(ctx, pvc)
```

Comment:
2 separate VR reconciles could update the annotation at the same time. (2 go routines etc.). This may need some rudimentary locking such that there is only one updater (fetch and update sequence) for a given PVC.

This should be quite rare, but something to keep track of.

Reply:
Agree but am thinking if 2 VR acts on the same PVC to add annotation, either one will fail and one will pass and should be fine. what do you think? as its a quite rare condition we can see how to fix it.